src/include/column.i

```diff
@@ -79,9 +81,11 @@ __col_insert_search_lt(WT_INSERT_HEAD *ins_head, uint64_t recno)
 	 * go as far as possible at each level before stepping down to the next.
 	 */
 	for (i = WT_SKIP_MAXDEPTH - 1, insp = &ins_head->head[i]; i >= 0;)
-		if (*insp != NULL && recno > WT_INSERT_RECNO(*insp)) {
-			ins = *insp;	/* GT: keep going at this level */
-			insp = &(*insp)->next[i];
+		if ((local_ins = *insp) != NULL &&
+		    recno > WT_INSERT_RECNO(*insp)) {
+			/* GT: keep going at this level */
+			ins = local_ins;
+			insp = &local_ins->next[i];
```

Comment:
I'm also a little concerned that in all 3 cases there's nothing to prevent the compiler from reading `*insp` twice on its own. Maybe:
```
ins = NULL;
for (i = WT_SKIP_MAXDEPTH - 1, insp = &ins_head->head[i]; i >= 0;)
        WT_ORDERED_READ(ins, *insp);
        if (ins != NULL && recno >= WT_INSERT_RECNO(ins)) {
                /* GTE: keep going at this level */
                insp = &local_ins->next[i];
        } else {
                --i;            /* LT: drop down a level */
                --insp;
        }
```